static/js/lib/site_content.ts

```diff
@@ -130,6 +130,43 @@ export const isSingletonCollectionItem = (
   configItem: BaseConfigItem
 ): configItem is SingletonConfigItem => "file" in configItem
 
+/**
+ * Creates a value for a payload to be sent to the server, for a given field. This is used by to
+ * contentFormValuesForPayload to figure out what should go in the payload for a field and nested fields.
+ */
```

Comment:
```suggestion
 * Returns a value for a payload to be sent to the server for a field, given a ConfigField
 * and the current state of the form (`values`).
 *
 * We need to look at the values for the entire form in order to handle
 * the case of Object fields, where for the 'nested' entries in the 
 * Object we need to look at the value for their 'parent' field in order
 * to set them correctly.
 */
```

I was thinking something like this, provided you think it makes sense. It's just been a little confusing to me in this file to have a series of functions that do fairly closely related things, so I want to make sure we name things well and keep track of what everything is supposed to be doing somehow.

Reply:
Sounds good